geopandas/io/file.py

```diff
@@ -796,3 +796,34 @@ def _geometry_types(df):
         geom_types = geom_types[0]
 
     return geom_types
+
+
+def _list_layers(filename) -> pd.DataFrame:
+    """List layers available in a file.
+
+    Provides an overview of layers available in a file or URL toghether with their
+    geometry types. When supported, this includes both spatial and non-spatial layers.
+    Non-spatial layers are indicated by the ``"geometry_type"`` column being ``None``.
+    GeoPandas will not read such layers but they can be read into pd.DataFrame using
+    :func:`pyogrio.read_dataframe`.
+
+    Parameters
+    ----------
+    filename : str, path object or file-like object
+        Either the absolute or relative path to the file or URL to
+        be opened, or any object with a read() method (such as an open file
+        or StringIO)
+
+    Returns
+    -------
+    pandas.DataFrame
+        A DataFrame with columns "name" and "geometry_type" and one row per layer.
+    """
```

Comment:
Is it worth adding an example of how to get just the layer names as a list?

E.g., 
`geopandas.list_layers(<path>).name.tolist()`

The DataFrame return seems like the cleanest way to handle the pairs of layer name and geometry type, but I wonder if folks will sometimes just want the layer names.